### PR TITLE
add SpeakerLabelHDFFromBlissJob

### PR DESCRIPTION
**Pull request by JackTemaki**
I had a local copy of the `SimpleHDFWriter` before, but this is really not clean. So now I decided to just import it from RETURNN, either directly if installed in the env or via a root path that is linked dynamically.

### CODE REVIEW

**Review by Atticus1806 — APPROVED** [review]

File: returnn/dataset.py [hunk 1]
```diff
@@ -90,3 +93,56 @@ def run(self):
...
+        )
+        hdf_writer = SimpleHDFWriter(
+            self.out_speaker_hdf.get_path(), dim=num_speakers, ndim=1
+        )
+
+        for recording in bliss.all_recordings():
+            for segment in recording.segments:
+                speaker_name = segment.speaker_name or recording.speaker_name
+                speaker_index = index_by_speaker[speaker_name]
+                segment_name = "/".join([bliss.name, recording.name, segment.name])
```
Comment (michelwi) [inline returnn/dataset.py:143]: what if there are sub-corpora? Better use the existing fullname function

```suggestion
                segment_name = segment.fullname()
```

File: returnn/dataset.py [hunk 2]
```diff
@@ -90,3 +93,56 @@ def run(self):
...
             self.out_std_dev.set(numpy.sqrt(total_var))
+
+
+class SpeakerLabelHDFFromBlissJob(Job):
+    """
+    Extract speakers from a bliss corpus and create an HDF file with a speaker index
+    matching the speaker entry in the corpus speakers for each segment
+    """
+
+    def __init__(self, bliss_corpus, returnn_root=None):
```
Comment (christophmluscher) [inline returnn/dataset.py:104]: typing

File: returnn/dataset.py [hunk 3]
```diff
@@ -90,3 +93,56 @@ def run(self):
...
+        """
+        :param bliss_corpus: bliss XML corpus where the speakers and segments are taken from
+        :param returnn_root: used to import SimpleHDFWriter from a specific path if not installed in the worker env
+        """
+        self.bliss_corpus = bliss_corpus
+        self.returnn_root = returnn_root
+        self.out_speaker_hdf = self.output_path("speaker_labels.hdf")
```
Comment (christophmluscher) [inline returnn/dataset.py:111]: add: `cached=True`?

File: returnn/dataset.py [hunk 4]
```diff
@@ -90,3 +93,56 @@ def run(self):
...
+        self.out_num_speakers = self.output_var("num_speakers")
+        self.out_speaker_dict = self.output_path("speaker_dict.pkl")
```
Comment (christophmluscher) [inline returnn/dataset.py:113]: add: `cached=True`?

**Review by christophmluscher — COMMENTED** [review]
more general question maybe: should all jobs use a temp file/dir?? should sisyphus handle this?

File: returnn/dataset.py (hunk 3 above)
Comment (JackTemaki) [reply returnn/dataset.py:111]: As we discussed somewhere else the Sisyphus internal caching only works for rasr related things, and those here are not. So I think there is no need. HDF caching will be done by RETURNN anyways.

File: returnn/dataset.py (hunk 2 above)
Comment (JackTemaki) [reply returnn/dataset.py:104]: Ah yes sure.

File: returnn/dataset.py (hunk 1 above)
Comment (JackTemaki) [reply returnn/dataset.py]: good catch

**Comment by JackTemaki** [comment]
> more general question maybe: should all jobs use a temp file/dir?? should sisyphus handle this?

In this case it is writing so fast that it makes no difference if we write to the NFS or first to a tempdir and then copy, but yes we can discuss this. I would not let sisyphus handle this per default, because then you have no intermediate work files if there is a crash.

**Review by michelwi — APPROVED** [review]

**Comment by christophmluscher** [comment]
> > more general question maybe: should all jobs use a temp file/dir?? should sisyphus handle this?
> 
> In this case it is writing so fast that **_I think_** it makes no difference if we write to the NFS or first to a tempdir and then copy, but yes we can discuss this. I would not let sisyphus handle this per default, because then you have no intermediate work files if there is a crash.

I agree that this is a more general discussion

**Review by christophmluscher — APPROVED** [review]

**Pull request merged**